pinot-segment-local/src/main/java/org/apache/pinot/segment/local/function/GroovyFunctionEvaluator.java

```diff
@@ -86,10 +86,6 @@ public List<String> getArguments() {
   public Object evaluate(GenericRow genericRow) {
     for (String argument : _arguments) {
       Object value = genericRow.getValue(argument);
-      if (value == null) {
```

Comment:
What is the behavior if the provided function doesn't handle `null` values? it might throw exception and is that handled?

Reply:
+1. We need to double check on what `_script.run()` would return for NullPointerException in the groovy side and we should handle it properly (e.g. on exception, we put `null` value. Or, we can even throw the exception and fail the segment creation.)

Reply:
after the groovy transform is run, the null value transform will be run, and if any value still remains null, NullValueTransformer will set the default value based on datatype. We run all these transformations in sequence.
 ```  
return new CompositeTransformer(Arrays
        .asList(new ExpressionTransformer(tableConfig, schema), new FilterTransformer(tableConfig),
            new DataTypeTransformer(schema), new NullValueTransformer(tableConfig, schema),
            new SanitizationTransformer(schema)));
```

Reply:
I think the suggestion here is to do a try-catch around `_script.run()` and return `null` if it throws exception

Reply:
Added the check